classes/class-wc-connect-paypal-ec.php

```diff
@@ -0,0 +1,250 @@
+		/**
+		 * Limit supported payment gateway features to payments
+		 */
+		public function supports( $supported, $feature, $gateway ) {
+			return 'ppec_paypal' === $gateway->id ? 'products' === $feature : $supported;
+		}
+
+		public function maybe_set_banner( $order_id ) {
+			$order          = wc_get_order( $order_id );
+			$payment_method = WC_Connect_Compatibility::instance()->get_payment_method( $order );
+
+			if ( 'ppec_paypal' === $payment_method ) {
+				update_option( 'wc_connect_banner_ppec', 'yes' );
+			}
+		}
+
+		/**
+		 * Once a payment is received, show prompt to connect a PayPal account on certain screens
+		 */
+		public function maybe_init_banner() {
+			if (
+				'yes' !== get_option( 'wc_connect_banner_ppec', null ) ||
+				'yes' === get_option( 'wc_connect_dismiss_banner_ppec', null )
+			) {
+				return;
+			}
+
+			$screen = get_current_screen();
+			if ( // Display if on any of these admin pages.
+				( // Orders list.
+					'shop_order' === $screen->post_type
+					&& 'edit' === $screen->base
+					)
+				|| ( // Edit order page.
+					'shop_order' === $screen->post_type
+					&& 'post' === $screen->base
+					)
+				|| ( // WooCommerce settings.
+					'woocommerce_page_wc-settings' === $screen->base
+					)
+				|| ( // WooCommerce featured extension page
+					'woocommerce_page_wc-addons' === $screen->base
+					&& isset( $_GET['section'] ) && 'featured' === $_GET['section']
+					)
+				|| ( // WooCommerce payment gateway extension page
+					'woocommerce_page_wc-addons' === $screen->base
+					&& isset( $_GET['section'] ) && 'payment_gateways' === $_GET['section']
+					)
+				|| 'plugins' === $screen->base
+			) {
+				wp_enqueue_style( 'wc_connect_banner' );
+				add_action( 'admin_notices', array( $this, 'banner' ) );
+			}
+		}
+
+		public function banner() {
+			$prompt = __( 'Link a new or existing PayPal account to enable PayPal Express Checkout features beyond simply taking payments: issue refunds, capture charges after order completion, and more.', 'woocommerce-services' );
+
+			$this->nux->show_nux_banner( array(
+				'title'          => __( 'Connect a PayPal account', 'woocommerce-services' ),
+				'description'    => esc_html( $prompt ),
+				'button_text'    => __( 'Connect', 'woocommerce-services' ),
+				'button_link'    => wc_gateway_ppec()->ips->get_signup_url( 'live' ),
+				'image_url'      => plugins_url( 'images/cashier.svg', dirname( __FILE__ ) ),
+				'should_show_jp' => false,
+				'dismiss_option' => 'ppec',
+			) );
+		}
+
+		public function initialize_settings() {
+			$settings = get_option( 'woocommerce_ppec_paypal_settings', array() );
+
+			if ( ! isset( $settings['button_size'] ) ) {
+				$gateway = new WC_Gateway_PPEC_With_PayPal();
+				$settings_meta = $gateway->form_fields;
+				foreach ( $settings_meta as $key => $setting_meta ) {
+					if ( ! isset( $settings[ $key ] ) && isset( $setting_meta['default'] ) ) {
+						$settings[ $key ] = $setting_meta['default'];
+					}
+				}
```

Comment:
Is this loop necessary? Could you use [`WC_Settings_API::process_admin_options()`](https://github.com/woocommerce/woocommerce/blob/de71c92fa0bf11de78e1a09b5a2fe1d53dfcf983/includes/abstracts/abstract-wc-settings-api.php#L174) instead?

Reply:
Ah, hopefully — will look into it.

Reply:
I believe it does what we need, but I seem to have run into trouble constructing a new payment gateway instance at this point in order to run that method — I can look further into how to make that work another time.